core/src/main/java/com/amplifyframework/devmenu/EnvironmentInfo.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.devmenu;
+
+import android.content.Context;
+import android.content.res.Resources;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.category.Category;
+import com.amplifyframework.core.plugin.Plugin;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Scanner;
+
+/**
+ * Contains information about the plugin versions added to the application
+ * and the developer's environment.
+ */
+public final class EnvironmentInfo {
+    // Name of the file that contains the developer environment information.
+    private static final String DEV_ENV_INFO_FILE_NAME = "localenvinfo";
```

Comment:
The file local-env-info.json currently resides in the amplify/.config directory in a customer's app project. Developer environment information will eventually be stored in this file and a copy of the file (named localenvinfo.json due to Android resource file naming restrictions) will be added to the app/src/main/res/raw directory in order to retrieve the information for the developer menu.

Reply:
^ I think it would be preferable to just make a new config block inside of our existing `amplifyconfiguration.json`. We _can't_ read the host-side file, and we probably won't benefit from having two separate `.json` files littered about the `src/main/res/raw`. This one should get design reviewed, ultimately. lizeyu@ is actively working on this so let's loop him in for review.

Reply:
@jamesonwilliams: Erica, Daniel, Abhash, and I already did a design review with Jake Litwicki where it was decided the local env info file the CLI team is already generating is the correct one for this purpose and this is what iOS has implemented and has the user importing as part of the setup process. The problem with using amplifyconfiguration.json is that it would lead to conflicts when developers make commits from different machines.

Reply:
> The problem with using amplifyconfiguration.json is that it would lead to conflicts when developers make commits from different machines with different environment info.

Very good point. 

Is there a design doc available to understand this a little more?    I'm curious to know how `localenvinfo.json` is generated.  Is it at compile time?

Reply:
There is a design doc, I just shared it with you.

Reply:
From the design doc, I can see that `local-env-info.json` already exists, and is generated on `amplify configure`.    However, there's still several questions that don't seem to be addressed in the doc.  

1. How/when is `local-env-info.json` copied to `localenvinfo.json` in the res/raw folder?
2. How is it kept up to date when node version or amplify-cli version on my machine changes?  I would say that incorrect environment info is probably worse than no info at all.
3. Since`local-env-info.json` is not intended to be under source control, how it is added to `.gitignore`?  Or is this something developers need to do manually?

Reply:
Questions answered via Chime:

> 1. How/when is local-env-info.json copied to localenvinfo.json in the res/raw folder?

The CLI will be updated to copy this file.  TBD on which command will trigger this.  Possibly `amplify push`.

> 2. How is it kept up to date when node version or amplify-cli version on my machine changes? I would say that incorrect environment info is probably worse than no info at all.

The goal is not to keep it up to date, but to make it match the environment info that was used to generate the Amplify project, since that is most useful for debugging.  

> 3. Since `local-env-info.json` is not intended to be under source control, how it is added to .gitignore? Or is this something developers need to do manually?

The CLI will be updated to add it to .gitignore when it is copied to the `res/raw` folder.